Snakefile

```diff
@@ -420,92 +637,1344 @@ rule atlas_experiment_summary:
+    """
+    Fetches metadata about the analysis methods used in ISL/iRap to preprocess the experiment,
+    to generate analysis methods.
+    """
+    conda: "envs/perl-atlas-modules.yaml"
+    log: "logs/{accession}-generate_methods_baseline_rnaseq.log"
+    params:
+        organism=get_organism(),
+        template=get_methods_template_baseline(),
+        isl_dir=get_isl_dir(),
+        isl_genomes=get_isl_genomes(),
+        irap_versions=get_irap_versions(),
+        irap_container=get_irap_container()
+    output:
+        methods=temp("{accession}-analysis-methods.tsv_baseline_rnaseq")
+    shell:
+        """
+        set -e # snakemake on the cluster doesn't stop on error when --keep-going is set
+        exec &> "{log}"
+
+        if [ ! -s {params.template} ] ; then
+            echo "Methods template not found "
+            exit 1
+        fi
+
+        source {workflow.basedir}/bin/reprocessing_routines.sh
+        expIslDir={params.isl_dir}/{wildcards.accession}/{params.organism}
+        echo "ISL dir: $expIslDir"
+        echo "ISL genome references: {params.isl_genomes}"
+
+        [ ! -z $expIslDir+x ] || (echo "snakemake param exp_isl_dir needs to defined in rule" && exit 1)
+
+        # IRAP methods version file
+        if [ ! -s "$expIslDir/irap.versions.tsv" ] ; then
+            echo "$expIslDir/irap.versions.tsv not found for {wildcards.accession} "
+            exit 1
+        fi
+
+        # set env variables for mapper and quantification methods from used in irap.
+        get_methods_from_irap "$expIslDir/irap.versions.tsv"
+        [ ! -z ${{baseline_mapper+x}} ] || (echo "Env var baseline_mapper not defined." && exit 1)
+        [ ! -z ${{baseline_quantMethod+x}} ] || (echo "Env var baseline_quantMethod not defined." && exit 1)
+        [ ! -z ${{de_mapper+x}} ] || (echo "Env var de_mapper not defined." && exit 1)
+        [ ! -z ${{de_quantMethod+x}} ] || (echo "Env var de_mapper not defined." && exit 1)
+        [ ! -z ${{de_deMethod+x}} ] || (echo "Env var de_mapper not defined." && exit 1)
+
+        echo $baseline_mapper
+        echo $baseline_quantMethod
+        echo $de_mapper
+        echo $de_quantMethod
+        echo $de_deMethod
+
+        # not used, only for differential rnaseq
+        deseq2version='none'
+        echo $deseq2version
+
+        echo "Cheking if IRAP versions file: {params.irap_versions} exists.."
+
+        if [ ! -s {params.irap_versions} ] ; then
+            echo "it does not - attempting to transfer the file from the singularity container {params.irap_container}"
+            singularity exec {params.irap_container} cp /opt/irap/aux/mk/irap_versions.mk {params.irap_versions}
+            if [ ! -s {params.irap_versions} ] ; then
+                echo "ERROR: Failed to retrieve {params.irap_versions} from singularity" >&2
+                exit 1
+            fi
+        else
+            echo "The file {params.irap_versions} already exists."
+        fi
```

Comment:
Done 61a9fd1. This logic was necessary to avoid:

> Not all output, log and benchmark files of rule get_irap_versions_file contain the same wildcards. This is crucial though, in order to avoid that two or more jobs write to the same file.